litellm/proxy/litellm_pre_call_utils.py

```diff
@@ -533,8 +533,14 @@ async def add_litellm_data_to_request(  # noqa: PLR0915
     if data.get(_metadata_variable_name, None) is None:
         data[_metadata_variable_name] = {}
 
-    # We want to log the "metadata" from the client side request. Avoid circular reference by not directly assigning metadata to itself.
+    # Parse metadata if it's a string (e.g., from multipart/form-data)
     if "metadata" in data and data["metadata"] is not None:
+        if isinstance(data["metadata"], str):
+            import json
+            try:
```

Comment:
create a helper in litellm core utils 

safe json loads 

(we have one for safe json dumps)